freedomofpress/securedrop-client
# provides tooling for interactive localization updates

# Description

Closes #1317 by:

1. removing the `l10n` GitHub Actions workflow added in #1282;
2. renaming the Make targets `extract-strings` and `check-strings` for clarity;
3. using `check-strings` in CI to "lint" that the translation catalog has been updated via `extract-strings`;
4. offering a Git pre-commit hook for `check-strings` configurable with `make hooks`; and
5. removing Make targets whose functions can be performed by Weblate add-ons without custom code or tooling.

Step (5) is this pull request's *addition* over "option A" #1347.

## How to think about this pull request

The following gist offers background considerations shared with, and for evaluating this "option B" pull request against, "option A" #1347:

https://gist.github.com/cfm/d3173206f55d7e44152928fcfd215031

# Test Plan

## To test the developer experience

1. [x] Change a `_()`-wrapped source string and commit it
2. [x] Push to testing branch like `1348-cfm-testing` (where `cfm` is your GitHub username ;-)
3. [x] Confirm that `test-buster` fails on `make check-strings` with the following error message:
    > Translation catalog is out of date. Please run "make extract-strings" and commit the changes.
4. [x] Run `make hooks` (or rerun `make venv` or your platform-appropriate variant)
5. [x] Try to `git commit --all` and confirm that the pre-commit hook fails on `make check-strings` as above
6. [x] Run `make extract-strings`
7. [x] Retry `git commit --all` and confirm that the commit succeeds
8. [x] Push to your testing branch
9. [x] Confirm that `test-buster` passes

## To test the Weblate features

1. [x] Ask @cfm to add you as a collaborator on `cfm/securedrop-client`
2. [x] `git remote add cfm git@github.com:cfm/securedrop-client.git`
3. [x] `git push -f cfm  1348-cfm-testing^:i18n-testing` (where `1348-cfm-testing` is the testing branch you used above), to push your string change *before* `make extract-strings`
4. [x] Reset Weblate from `cfm/securedrop-client`: https://weblate-sandbox.securedrop.org/projects/securedrop/securedrop-client/#repository
5. [x] ~Explore Weblate and the pull request it maintains in `cfm/securedrop-client`~
6. [x] `git push cfm  1348-cfm-testing:i18n-testing` (where `1348-cfm-testing` is the testing branch you used above), to push your string change *with* `make extract-strings`
7. [x] Update Weblate from `cfm/securedrop-client`: https://weblate-sandbox.securedrop.org/projects/securedrop/securedrop-client/#repository
8. [x] Continue exploring Weblate and `cfm/securedrop-client`
    1. [x] Translate the string you changed
    2. [x] Push from Weblate: https://weblate-sandbox.securedrop.org/projects/securedrop/securedrop-client/#repository

# Checklist

## Before merge

- [x] Remove `(option B)` from title and close #1347 without merging

## After merge

- [ ] Replicate add-on configuration for the `securedrop-client` Weblate component from `weblate-sandbox` to `weblate`:

![image](https://user-images.githubusercontent.com/357435/144148302-0e398412-f02a-46b9-a068-1479e958296b.png)

## For review

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - I have tested these changes in the appropriate Qubes environment
 - I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - I have written a migration and upgraded a test database based on `main` and confirmed that the migration applies cleanly
 - I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - I need help writing a database migration
